src/core/snippets/snippets.ts

```diff
@@ -31,48 +74,60 @@ export class UIPSnippets extends UIPPlugin {
   }
 
   protected bindEvents() {
-    this.addEventListener('click', this._onClick);
+    this.$snippetList.addEventListener('click', this._onListClick);
+    this.$dropdownControl.addEventListener('click', this._onDropdownClick);
   }
 
   protected unbindEvents() {
-    this.removeEventListener('click', this._onClick);
+    this.$snippetList.removeEventListener('click', this._onListClick);
+    this.$dropdownControl.removeEventListener('click', this._onDropdownClick);
   }
 
   @memoize()
   public get $items(): HTMLElement[] {
-    const items = this.querySelectorAll(`.${UIPSnippets.ITEM_CLASS}`);
+    const items = this.querySelectorAll(`.${UIPSnippets.LIST_ITEM}`);
     return Array.from(items) as HTMLElement[];
   }
 
   public get $active(): HTMLElement | null {
-    return this.$items.find(item => item.classList.contains(UIPSnippets.ACTIVE_CLASS)) || null;
+    return this.$items.find(item => item.classList.contains(UIPSnippets.ACTIVE_ITEM)) || null;
   }
   public set $active(snippet: HTMLElement | null) {
-    this.$items.forEach((item) => item.classList.toggle(UIPSnippets.ACTIVE_CLASS, snippet === item));
-    this.applyActive();
+    this.$items.forEach((item) => item.classList.toggle(UIPSnippets.ACTIVE_ITEM, snippet === item));
   }
 
   @memoize()
   public get $scroll() {
     const $scroll = document.createElement('esl-scrollbar');
-    $scroll.setAttribute('target', '::prev(.snippets-list)');
+    $scroll.setAttribute('target', '::prev');
     return $scroll;
   }
 
   /** Render snippets list. */
   protected render(): void {
-    const snippets = this.querySelectorAll(UIPSnippets.CONTENT_SEL);
+    const snippets = this.model!.snippets;
     if (!snippets.length) return;
-    const $ul = document.createElement('ul');
-    $ul.className = 'snippets-list esl-scrollable-content';
+
+    snippets.length > 1
+      ? this.renderDropdown(snippets, this.$title)
+      : this.appendChild(this.$title);
+  }
+
+  protected renderDropdown(snippets: HTMLTemplateElement[], title: HTMLElement) {
```

Comment:
snippets are already an array, so there is no need to do `Array.from(snippets)` once again

Reply:
updated